### PR TITLE
Fix sonar issue: Remove Optional null check in ResultIterator

### PR DESCRIPTION
**Pull request by sandynz**
Changes proposed in this pull request:
  - Fix sonar issue: Remove Optional null check in ResultIterator

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by azexcy — APPROVED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java [hunk 1]
```diff
@@ -65,6 +66,8 @@ public Iterator<DataConsistencyCalculatedResult> iterator() {
     @RequiredArgsConstructor
```
Comment (azexcy) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java:66]: ci failed

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java
Comment (azexcy) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java:1]: Ci failed

Comment (azexcy) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java:1]: Ci failed

**Review by azexcy — CHANGES_REQUESTED** [review]

**Review by azexcy — APPROVED** [review]

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java
Comment (sandynz) [reply kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/algorithm/AbstractStreamingDataConsistencyCalculateAlgorithm.java:1]: There's random error in infra module and pipeline e2e, it works after rerun. We need to check it later.

**Pull request merged**